src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -584,5 +589,1762 @@ void M()
+    {
+    }
+}
+class C1
+{
+    void M1()
+    {
+        new object().[|ExtensionM1|](1);
+    }
+}}";
+            var fix =
+@"
+namespace N {
+static class Extensions
+{
+    public static void ExtensionM1(this object o, int v)
+    {
+    }
+}
+class C1
+{
+    void M1()
+    {
+        new object().ExtensionM1(1);
+    }
+}}";
+            await TestInRegularAndScriptAsync(code, fix);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationExtensionMethod_StaticInvocationStyle()
+        {
+            // error CS1501: No overload for method 'ExtensionM1' takes 2 arguments
+            var code =
+@"
+namespace N {
+static class Extensions
+{
+    public static void ExtensionM1(this object o)
+    {
+    }
+}
+class C1
+{
+    void M1()
+    {
+        Extensions.[|ExtensionM1|](new object(), 1);
+    }
+}}";
+            var fix =
+@"
+namespace N {
+static class Extensions
+{
+    public static void ExtensionM1(this object o, int v)
+    {
+    }
+}
+class C1
+{
+    void M1()
+    {
+        Extensions.ExtensionM1(new object(), 1);
+    }
+}}";
+            await TestInRegularAndScriptAsync(code, fix);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationOverride()
+        {
+            var code = @"
+class Base
+{
+    protected virtual void M1() { }
+}
+class C1 : Base
+{
+    protected override void M1() { }
+    void M2()
+    {
+        [|M1|](1);
+    }
+}";
+            var fix_DeclarationOnly = @"
+class Base
+{
+    protected virtual void M1() { }
+}
+class C1 : Base
+{
+    protected override void M1(int v) { }
+    void M2()
+    {
+        M1(1);
+    }
+}";
+            var fix_All = @"
+class Base
+{
+    protected virtual void M1(int v) { }
+}
+class C1 : Base
+{
+    protected override void M1(int v) { }
+    void M2()
+    {
+        M1(1);
+    }
+}";
+            await TestInRegularAndScriptAsync(code, fix_DeclarationOnly, index: 0);
+            await TestInRegularAndScriptAsync(code, fix_All, index: 1);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationExplicitInterface()
+        {
+            var code = @"
+interface I1
+{
+    void M1();
+}
+class C1 : I1
+{
+    void I1.M1() { }
+    void M2()
+    {
+        ((I1)this).[|M1|](1);
+    }
+}";
+            var fix_DeclarationOnly = @"
+interface I1
+{
+    void M1(int v);
+}
+class C1 : I1
+{
+    void I1.M1() { }
+    void M2()
+    {
+        ((I1)this).M1(1);
+    }
+}";
+            var fix_All = @"
+interface I1
+{
+    void M1(int v);
+}
+class C1 : I1
+{
+    void I1.M1(int v) { }
```

Comment:
so it fix up all implementations of interface I1

Reply:
curious, what case made you to decide to add "declaration only fix" when you have "fix all" implemented.

Reply:
I think the reasoning behind the two offerings is that adding a parameter is a breaking change. This is different from *change signature*. Change signature does not change the behavior of the code and does not introduces compiler errors. Adding a parameter does both often. Having two offerings the user can decide whether he want's to add the parameter to just one method at a time (to have a look how many call sites are breaking and what else isn't working anymore) or he can go all in and fix several declarations at once. In both cases he needs to fix compiler errors after the fix.

I discussed this with @CyrusNajmabadi back in January. I approached the problem by ignoring any related declarations (my first attempt didn't include any cascading and I suggested to not do any cascading). @CyrusNajmabadi weighed in and he said that cascading is what users would expect and what is more in line with the current behavior of other fixes. But because of the many different ways to deal with the cascading problem (there are many ways to fix the errors introduced by this fix, e.g. you can remove the `override` keyword from a method or you can change the `virtual` method or you can introduce a new `virtual` overload, and so on), we decided to offer both fixes. @CyrusNajmabadi, I hope I got your arguments right because the discussion we had was brief and is a few month back.

Reply:
sure. I see the reasoning and fine to offer 2. but probably we want other people's opinion as well to see whether they have different opinions such as offering just 1 is enough (cascading case if it exists otherwise, non-cascading). 

...

whether it is doing exact same thing as existing signature change features or not, sure it dos different behavior since it is different features. but I believe 2 does conceptually same thing whether it introduce error or not, so I believe two should behave similar. anyway, its my opinion, so wants to discuss with others. but not saying it should block your PR.

...

one of reason I believe it might be better to change all references is that, 

```csharp
class C
    {
        void Method(int i) { }
        void Method(long l) { }

        void Test()
        {
            Method(1); <= add "s" here
            Method(2);
        }
    }
```
if you change Method(int i, string s) to add new parameter "s", Method(2) now bound to Method(long l), so it makes harder to find such thing after. 

even without that, just it is just more convenient for users if it just adds "default" as argument to new parameters I think.